src/odemis/driver/piezomotor.py

```diff
@@ -0,0 +1,453 @@
+        self.setWaveform(WAVEFORM_PARK)  # power off
+        self._serial.close()
+
+    def _findDevice(self, port, address=None):
+        """
+        Look for a compatible device
+        port (str): pattern for the port name
+        address (None or int): the address of the
+        return (serial, int): the (opened) serial port used, and the actual address
+        raises:
+            IOError: if no device are found
+        """
+        if port.startswith("/dev/fake"):
+            names = [port]
+        elif os.name == "nt":
+            raise NotImplementedError("Windows not supported")
+        else:
+            names = glob.glob(port)
+
+        for n in names:
+            try:
+                serial = self._openSerialPort(n)
+            except IOError as ex:
+                # not possible to use this port? next one!
+                logging.info("Skipping port %s, which is not available (%s)", n, ex)
+                continue
+
+            # check whether it answers with the right address
+            try:
+                # If any garbage was previously received, make it discarded.
+                self._serial = serial
+                self._serial.flush()
+                if 'pmd' in self.getVersion().lower():
+                    return serial  # found it!
+            except Exception as ex:
+                logging.debug("Port %s doesn't seem to have a TMCM device connected: %s",
+                              n, ex)
+            serial.close()  # make sure to close/unlock that port
+        else:
+            raise IOError("Failed to find a TMCM controller on ports '%s'. "
+                          "Check that the device is turned on and "
+                          "connected to the computer." % (port,))
+
+
+    @staticmethod
+    def _openSerialPort(port):
+        """
+        Opens the given serial port the right way for a Thorlabs APT device.
+        port (string): the name of the serial port (e.g., /dev/ttyUSB0)
+        return (serial): the opened serial port
+        raise HwError: if the serial port cannot be opened (doesn't exist, or
+          already opened)
+        """
+        # For debugging purpose
+        if port == "/dev/fake":
+            return PMDSimulator(timeout=0.1)
+
+        try:
+            ser = serial.Serial(
+                port=port,
+                baudrate=BAUDRATE,
+                bytesize=serial.EIGHTBITS,
+                parity=serial.PARITY_NONE,
+                stopbits=serial.STOPBITS_ONE,
+                timeout=0.3,  # s
+            )
+        except:
+            raise IOError("Failed to find a TMCM controller on port '%s'. "
+                          "Check that the device is turned on and "
+                          "connected to the computer." % (port,))
+
+        return ser
+
+    def set_axis_address(self, new_address, old_address=0):
+        """
+        Set the address of the axis. The factory default is 0 for all boards. Don't use this
```

Comment:
I think this command should take current the axis number, the new address. Or maybe just make it generic setParam(axis, param, value).